code/dce-downloader/python/dce.py

```diff
@@ -11,78 +11,115 @@
 import requests
 import logging
 import json
+import zlib
 
 #logging.basicConfig(level=logging.DEBUG)
 
-# host url for DCE services
-DCE_HOSTS = {
-    "stg": "data-stg.nexus.bazaarvoice.com",
-    "prod": "data.nexus.bazaarvoice.com"
-}
+# Create hmac signature
+def createSignature(passkey, secretKey, timestamp, path):
+    # If 'path' parameter will be in the GET request, we must include it as part of the HMAC signature.
+
+    if path:
+        message = "path={0}&passkey={1}&timestamp={2}".format(path, passkey, timestamp)
+    else:
+        message = "passkey={0}&timestamp={1}".format(passkey, timestamp)
+
+    return hmac.new(secretKey, message, hashlib.sha256).hexdigest()
+
+def buildBVHeaders(passkey, signature, timestamp):
+    return {'X-Bazaarvoice-Passkey': passkey, 'X-Bazaarvoice-Signature': signature, 'X-Bazaarvoice-Timestamp': timestamp}
+
+def doHttpGet(url, passkey, secretKey, path):
+    timestamp = str(round(time.time() * 1000))
+
+    # Get current manifests
+    signature = createSignature(passkey=passkey, secretKey=secretKey, timestamp=timestamp, path=path)
+
+    headers = buildBVHeaders(passkey, signature, timestamp)
+
+    params = { 'path' : path } if path else {}
+
+    if path:
+        print "Downloading " + path;
+    else:
+        print "Retrieving manifests"
+
+    resp = requests.get(url, params=params, headers=headers, timeout=60, allow_redirects=True, stream=True)
+    return resp
+
+def saveFile(dest, path, content):
+    file = dest + path
+    print "Saving as " + file
+
+    dirname = os.path.dirname(file)
+    if not os.path.exists(dirname):
+        os.makedirs(dirname)
+
+    with open(file, "wb") as file:
+        file.write(content)
+
 
 # Main part
 if __name__ == '__main__':
     # Setting script parameters and variables
     p = argparse.ArgumentParser(description=__doc__, formatter_class=argparse.RawDescriptionHelpFormatter)
-    p.add_argument('-key-path',  dest='keypath', help='path to json file which includes x-api-key and shared key. '
-                                                      'Refer to keys.json, which is used if not specified')
-    p.add_argument('-env',  dest='environment', help='environment of DCE service, stg or prod.')
-    p.add_argument('-path', dest='path', help='path to target file')
+    p.add_argument('-config',  dest='configFile', help='path to configuration (default is ../config.json)')
+    p.add_argument('-env',  dest='environment', help='environment of DCE service (must be present in config file)')
+    p.add_argument('-path', dest='path', help='DCE file')
```

Comment:
We should probably have the flags be the same across Python and Node (e.g. "--env" instead of "-env"). Additionally, convention for full-name-length command line flags is to have double dashes.